deepmd/pt/model/model/ener_hess_model.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from copy import (
+    deepcopy,
+)
+from typing import (
+    Optional,
+)
+
+import torch
+
+from deepmd.pt.model.atomic_model import (
+    DPEnergyAtomicModel,
+)
+from deepmd.pt.model.model.model import (
+    BaseModel,
+)
+
+from .dp_model import (
+    DPModelCommon,
+)
+from .make_hessian_model import (
+    make_hessian_model,
+)
+from .make_model import (
+    make_model,
+)
+
+DPEnergyModel_ = make_model(DPEnergyAtomicModel)
+DPEnergyModel_ = make_hessian_model(DPEnergyModel_)
+
+
+@BaseModel.register("ener_hess")
+class EnergyHessianModel(DPModelCommon, DPEnergyModel_):
```

Comment:
I recommend we merging the `EnergyHessianModel` into `EnergyModel`, i.e. making a energy model by `make_hessian_model`, and control if the model outputs hessian by the method `requires_hessian`. 
- if `requires_hessian` is not set, the behavior of the returned object from `make_hessian_model` should be exactly the same as that from `make_model`.
- if  `requires_hessian` is set, the model gives hessian. 

The core idea is to control the model's output at **runtime**.

The advantage of the design is that we can easily switch from a standard energy model to hessian energy model and vice versa (imagine that we want to firstly train a model without hessian, and then finetune it with hessian), and do not need to maintain two implementations.

Reply:
I realized that the main difficulty is the jit. The hessian model is not jitable.

Reply:
Good suggestions for simplifying and optimizing the code structure. Now the `ener_hess_model.py` is removed and class `EnergyHessianModel` is merged to class `EnergyModel` in `ener_model.py` with Hessian calculation activated by an additional attribute.

Reply:
Since the Hessian model is not specified separately, the `jit` cannot be optionally declared during the model definition, so the `torch.script.jit` in `deep_eval.py` will look less elegant.